tests/test_api.py

```diff
@@ -3,98 +3,85 @@
 import logging
 import os
 import re
-import time
 from math import ceil
 from time import perf_counter
+from typing import Dict, Generator, List, Tuple, Union
 
-import numpy as np
+import flask
 import pandas as pd  # third party library import
 import pytest
+from flask.testing import FlaskClient
 
 from schematic.configuration.configuration import Configuration
-from schematic.schemas.data_model_parser import DataModelParser
 from schematic.schemas.data_model_graph import DataModelGraph, DataModelGraphExplorer
-from schematic.schemas.data_model_relationships import DataModelRelationships
-
+from schematic.schemas.data_model_parser import DataModelParser
 from schematic_api.api import create_app
 
-
 logging.basicConfig(level=logging.INFO)
 logger = logging.getLogger(__name__)
 
+BENCHMARK_DATA_MODEL_JSON_LD = "https://raw.githubusercontent.com/Sage-Bionetworks/schematic/develop/tests/data/example.single_rule.model.jsonld"
+DATA_MODEL_JSON_LD = "https://raw.githubusercontent.com/Sage-Bionetworks/schematic/develop/tests/data/example.model.jsonld"
+
 
-## TO DO: Clean up url and use a global variable SERVER_URL
 @pytest.fixture(scope="class")
-def app():
+def app() -> flask.Flask:
     app = create_app()
-    yield app
+    return app
 
 
 @pytest.fixture(scope="class")
-def client(app):
+def client(app: flask.Flask) -> Generator[FlaskClient, None, None]:
     app.config["SCHEMATIC_CONFIG"] = None
 
     with app.test_client() as client:
         yield client
 
 
 @pytest.fixture(scope="class")
-def test_manifest_csv(helpers):
+def test_manifest_csv(helpers) -> str:
     test_manifest_path = helpers.get_data_path("mock_manifests/Valid_Test_Manifest.csv")
-    yield test_manifest_path
+    return test_manifest_path
 
 
 @pytest.fixture(scope="class")
-def test_manifest_submit(helpers):
+def test_manifest_submit(helpers) -> str:
     test_manifest_path = helpers.get_data_path(
         "mock_manifests/example_biospecimen_test.csv"
     )
-    yield test_manifest_path
+    return test_manifest_path
```

Comment:
May I ask why switch to return?  The yield is like a "teardown" of resources

Reply:
yield is preferred if it needs to be context managed or there is code after the yield to clean up the resources. https://docs.pytest.org/en/stable/how-to/fixtures.html#teardown-cleanup-aka-fixture-finalization covers it pretty well. Since in this case we are doing neither of these it makes the return type of the function a `str` instead of a `Generator[str, None, None]`. Just a bit clearer. Though if there are other reasons to use yield I don't mind either way.